### PR TITLE
util/generate: Use the same template for all generations

### PR DESCRIPTION
**Pull request by dvermd**
This PR also do not repeat the test name creation.

Resolves: #679

### CODE REVIEW

File: util/exercise/src/cmd/generate.rs [hunk 1]
```diff
@@ -246,7 +230,33 @@ pub fn generate_exercise(exercise_name: &str, use_maplit: bool) -> Result<()> {
...
+                )
+                .into_bytes(),
+            )?;
+            if use_maplit {
+                test_file.write_all(b"#[macro_use]\nextern crate maplit;\n")?;
+            }
+
+            test_file.write_all(
+                &format!(
+                    "extern crate {escaped_exercise_name};\n\
```
Comment (coriolinus) [inline util/exercise/src/cmd/generate.rs]: As this track now uses the 2018 edition by default, we can remove the `extern crate` line.

Comment (dvermd) [reply util/exercise/src/cmd/generate.rs]: I'll remove that.
Following this advice, what about also changing 
```
#[macro_use]
extern crate maplit;
```
to 
`use maplit::*;`

**Comment by coriolinus** [comment]
It's not a bad idea to remove the extern crate, but we shouldn't be
encouraging people to use global imports. I think

use maplit::maplit;

will do what we need, but that'll need to be tested.

**Comment by dvermd** [comment]
A quick example from maplit shows that `use maplit::maplit;` is not working and one should write `use maplit::hashmap;` if the `hashmap!` macro is to be used.

Right now, there's 5 macros in the maplit crate. I might import them all explicitly and then `cargo build` will warn about unused imports.
I'm not happy with this but this is the best I can think right now.

Another point : should I remove one of the "[utility][utility]" in the template ? I didn't manage to find documentation on a specific behaviour related to "[]" in doc comments.

**Comment by coriolinus** [comment]
use maplit::hashmap;

is correct. That was my mistake earlier. I don't think the generator uses
any other literals, so we should not import the others.

You should not remove the "[utility][utility]"; that's a markdown external
hyperlink, which inlines the link which appears lower in the comment.

**Pull request merged**